pkg/util/hardware/hw_utils_test.go

```diff
@@ -39,6 +58,62 @@ var _ = Describe("Hardware utils test", func() {
 		})
 	})
 
+	Context("CPU threads grouping", func() {
```

Comment:
we aren't guaranteed to have siblings assigned to the pod. For example, we could request 4 cores and get 4 vcpus non of which are siblings, or some mix in between.

Can we get some tests that validate that scenario as well?

Reply:
I think the no-siblings use case is already covered.

```go
				// No siblings: expect res, nil
				{cpus: cpus8, siblings: siblings00, res: cpus8},
				{cpus: cpus3, siblings: siblings00, res: cpus3},
```

Added one more test scenario with mixed siblings/non-siblings cpus.